### PR TITLE
Expand support for creating new artifacts

### PR DESCRIPTION
**Pull request by lazappi**
This PR:

- [x] Adds a `from_path()` method to create artifacts from paths
- [x] Adds a `from_anndata()` method to create artifacts from `AnnData` objects

**Removed**

- [x] ~Adds a `from_dir()` method to create lists of artifacts from directories~
- [x] ~Adds a `Instance$save()` method which can save lists of records~

Checklist:

- [x] Add tests (partially)
- [x] Pass CI checks
- [x] Update man pages and regenerate
- [ ] Add new features to vignettes
- [x] Update architecture vignette
- [x] Update development vignette
- [x] Update features in `README`
- [x] Update `CHANGELOG`

### CODE REVIEW

**Comment by falexwolf** [comment]
Great! 🤩 

One question: Rather than calling it `from_file()`, should it maybe be called `from_path()`? Otherwise we also need `from_folder()` and that'd be very confusing with `from_dir()`.

If possible, to be honest, I'd hold off from adding `from_dir()` to the API -- we might even remove it from Python or rename it somehow because it's pretty confusing that it returns a list of artifacts rather than a single artifact or the whole directory.

File: R/Registry.R [hunk 1]
```diff
@@ -172,6 +172,102 @@ Registry <- R6::R6Class( # nolint object_name_linter
...
+    #' @param run A `Run` object that creates the record
+    #'
+    #' @details
+    #' Creating records is only possible for the default instance, requires the
+    #' Python `lamindb` module and is only implemented for the core `Artifact`
+    #' registry.
+    #'
+    #' @return A `TemporaryRecord` object containing the new record. This is not
+    #' saved to the database until `temp_record$save()` is called.
+    from_path = function(path, key = NULL, description = NULL, run = NULL) {
```
Comment (rcannood) [inline R/Registry.R:189]: This should only be added if the class is an Artifact Registry.

We probably need a

```r
create_registry <- function(instance, module, api, registry_name, registry_schema) {
  public <- list(
    initialize = function(instance, module, api, registry_name, registry_schema) {
      super$new(
        instance = instance,
        module = module,
        api = api,
        registry_name = registry_name,
        registry_schema = registry_schema
      )
    },
    # ...
  )

  if (registry_name == "Artifact") {
    public$from_path <- ...
    public$from_anndata <- ..
    public$from_dir <- ...
    public$from_mudata <- ...
  }
  if (registry_names == "FeatureSet") {
    public$from_values <- ...
  }
  # ...

  RichRegistryClass <- R6::R6Class(
    <some class name>,
    cloneable = FALSE,
    inherit = Registry,
    public = public
  )

  RichRegistryClass
}
```

Comment (rcannood) [reply R/Registry.R:189]: Or the other way around

```r
create_registry <- function(instance, module, api, registry_name, registry_schema) {
  public <- list(
    initialize = function(instance, module, api, registry_name, registry_schema) {
      super$new(
        instance = instance,
        module = module,
        api = api,
        registry_name = registry_name,
        registry_schema = registry_schema
      )
    },
    from_path = function(...) {
      ...
    },
    from_anndata = function(...) {
      ...
    },
    from_dir = function(...) {
      ...
    },
    # ...
  )

  if (registry_name != "Artifact") {
    public$from_path <- NULL
    public$from_anndata <- NULL
    public$from_dir <- NULL
    public$from_mudata <- NULL
  }
  if (registry_names != "FeatureSet") {
    public$from_values <- NULL
  }
  # ...

  RichRegistryClass <- R6::R6Class(
    paste0(registry_name, "Registry"),
    cloneable = FALSE,
    inherit = Registry,
    public = public
  )

  RichRegistryClass
}

Comment (rcannood) [reply R/Registry.R:189]: Lets tackle this in a separate PR!

**Pull request merged**